internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/integration/utils/KafkaBrokerFactory.java

```diff
@@ -139,7 +143,10 @@ private KafkaBrokerWrapper(
       this.sslPort = sslPort;
       this.zkServerWrapper = zkServerWrapper;
       this.shouldCloseZkServer = shouldCloseZkServer;
-      pubSubClientsFactory = new PubSubClientsFactory(new ApacheKafkaProducerAdapterFactory());
+      pubSubClientsFactory = new PubSubClientsFactory(
+          new ApacheKafkaProducerAdapterFactory(),
```

Comment:
I see this pattern is used everywhere. Why not just have a new PubSubClientsFactory() constructor that takes 0 argument and do the initialization here?

Reply:
The expectation of the `PubSubClientFactory` is to take different combination of PubSubAdapters for consumer, producer or admin. As for a specific component like controller, we may want to change just producer, this `PubSubClientFactory` can support hybrid clients for one component.